c++/src/kj/time.c++

```diff
@@ -268,6 +268,9 @@ CappedArray<char, sizeof(int64_t) * 3 + 2 + 4> KJ_STRINGIFY(Date d) {
   return kj::toCharSequence(d - UNIX_EPOCH);
 }
 CappedArray<char, sizeof(int64_t) * 3 + 2 + 4> KJ_STRINGIFY(Duration d) {
+  bool negative = d < 0 * kj::SECONDS;
+  if (negative) d = 0 * kj::SECONDS - d;
+
   auto digits = kj::toCharSequence(d / kj::NANOSECONDS);
```

Comment:
FWIW, it's not quite clear to me why `kj::toCharSequence()`'s existing handling of signed types isn't already adding a negative sign.  I'd expect `d / kj::NANOSECONDS` to yield an `int64_t`, which I would have thought would be covered by one of the existing `STRINGIFY_INT()` overloads.

Reply:
The problem is the logic below that chooses an appropriate unit doesn't account for the negative sign correctly.

Reply:
Ohhh... that makes more sense, thanks.